test/e2e/connectivity_test.go

```diff
@@ -136,34 +146,54 @@ func TestHostPortPodConnectivity(t *testing.T) {
 	data.testHostPortPodConnectivity(t)
 }
 
-// createPodsOnDifferentNodes creates numPods busybox test Pods and assign them to all the different
-// Nodes in round-robin fashion, then returns the names of the created Pods as well as a function
-// which will delete the Pods when called.
-func createPodsOnDifferentNodes(t *testing.T, data *TestData, numPods int) (podNames []string, cleanup func()) {
-	podNames = make([]string, 0, numPods)
+// createPodsOnDifferentNodes creates agnhost Pods through a DaemonSet. This function returns information of the created
+// Pods as well as a function which will delete the Pods when called.
```

Comment:
even though this is just used for testing, do you think you could add a quick comment about the `podInfos` slice, in particular how Pods are ordered within the slice?

Reply:
Good idea. Updated.